manifests/oss/kustomization.yaml

```diff
@@ -14,4 +14,5 @@
 
 resources:
 - ../base
+- ../hierarchyconfig-crd.yaml
```

Comment:
If this is going to be a common manifest, please move this to the base.

What is the analysis on whether this CRD needs to be installed on the cluster? It seems preferable to not install it on the cluster if there isn't a use case.

Reply:
> What is the analysis on whether this CRD needs to be installed on the cluster? It seems preferable to not install it on the cluster if there isn't a use case.

I don't have time to do the analysis. If the analysis is a blocker, someone else needs to own it.

Reply:
I think it largely depends on whether tools read CRDs from a cluster before validation or whether the tools import the CRDs/schemes through other means. But the list of tools we want to support would need to first be assembled.

Reply:
If we want to skip analysis and be defensive, we can just apply everything and let the user figure out client-side validation. But ofc, even that strategy might be confusing, since users might assume these resources should be or need to be applied to the cluster... It's kind of a lose-lose UX scenario.

Reply:
+1 on applying the CRD.

Reply:
Can you add a comment to the kustomize file that explains why the CRD is being included, even if the HCs should never be applied?

Reply:
Added a single-line comment.